Makefile

```diff
@@ -4,6 +4,9 @@ PREFIX?=$(shell pwd)
 # Used to populate version variable in main package.
 VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
 GO_LDFLAGS=-ldflags "-X `go list ./version`.Version $(VERSION)"
+ifeq (${REGISTRY_DEBUG},true)
+	GO_GCFLAGS=-gcflags "-N -l"
```

Comment:
These flags are passed to 6c and prevent inlining functions and storing
variables in registers, making go programs more debuggable in gdb.

On Wed, Jun 10, 2015 at 5:05 PM, Olivier Gambier <notifications@github.com>
wrote:

> In Makefile
> <https://github.com/docker/distribution/pull/616#discussion_r32179539>:
>
> > @@ -4,6 +4,9 @@ PREFIX?=$(shell pwd)
> >  # Used to populate version variable in main package.
> >  VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
> >  GO_LDFLAGS=-ldflags "-X `go list ./version`.Version $(VERSION)"
> > +ifeq (${REGISTRY_DEBUG},true)
> > +	GO_GCFLAGS=-gcflags "-N -l"
>
> What does -l do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/distribution/pull/616/files#r32179539>.
>